client/allocrunner/alloc_runner.go

```diff
@@ -178,38 +178,50 @@ func (ar *allocRunner) WaitCh() <-chan struct{} {
 
 // Run is the main goroutine that executes all the tasks.
```

Comment:
Should we document that callers should invoke it as a goroutine?  The comment already kinda says that, but this PR changes how it's being called.